geometry/signature_body.hpp

```diff
@@ -100,19 +104,33 @@ Signature<FromFrame, ToFrame>::operator()(
 }
 
 template<typename FromFrame, typename ToFrame>
-OrthogonalMap<FromFrame, ToFrame> Signature<FromFrame, ToFrame>::Forget()
-    const {
-  if (x_ == y_ && y_ == z_) {
-    return OrthogonalMap<FromFrame, ToFrame>(Quaternion(1));
+template<template<typename, typename> typename LinearMap>
+LinearMap<FromFrame, ToFrame> Signature<FromFrame, ToFrame>::Forget() const {
+  if constexpr (is_same_template_v<LinearMap, OrthogonalMap>) {
+    if (x_ == y_ && y_ == z_) {
+      return OrthogonalMap<FromFrame, ToFrame>(Quaternion(1));
+    }
+    // The signature is neither +++ nor ---, so dividing it by its determinant
+    // yields a 180° rotation around one of the axes (+--, -+-, or --+).
+    R3Element<double> const axis = (determinant_ * x_).is_positive()
+                                       ? R3Element<double>{1, 0, 0}
+                                       : (determinant_ * y_).is_positive()
+                                             ? R3Element<double>{0, 1, 0}
+                                             : R3Element<double>{0, 0, 1};
+    return OrthogonalMap<FromFrame, ToFrame>(Quaternion(0, axis));
+  } else if constexpr (is_same_template_v<LinearMap, Rotation>) {
+    if (x_ == y_ && y_ == z_) {
+      return Rotation<FromFrame, ToFrame>::Identity();
+    }
+    R3Element<double> const axis = (determinant_ * x_).is_positive()
+                                       ? R3Element<double>{1, 0, 0}
+                                       : (determinant_ * y_).is_positive()
+                                             ? R3Element<double>{0, 1, 0}
+                                             : R3Element<double>{0, 0, 1};
+    return Rotation<FromFrame, ToFrame>(Quaternion(0, axis));
```

Comment:
I think the code here could be generic (we construct either an `OrthogonalMap` or a `Rotation`, using the same quaternion), with no need for `if constexpr`.

A `static_assert` that `LinearMap` is either `OrthogonalMap` or `Rotation`, and that is is `Rotation` for equal handedness, can be used to make the error message friendlier.